atd-etl/cr3_download/cr3_download.py

```diff
@@ -5,145 +5,186 @@
-
-    crashes_list = response["data"]["atd_txdot_crashes"]
-
-    crashes_list_without_skips = [
-        x for x in crashes_list if x["crash_id"] not in known_skips
-    ]
-    print(
-        f"\nList of {len(crashes_list_without_skips)} crashes needing CR3 download: %s"
-        % json.dumps(crashes_list_without_skips)
+
+def load_secrets(one_password_client, vault_id):
+    """Load required secrets from 1Password."""
+    required_secrets = {
+        "HASURA_ENDPOINT": {
+            "opitem": "Vision Zero graphql-engine Endpoints",
+            "opfield": "production.GraphQL Endpoint",
+            "opvault": vault_id,
+        },
+        "HASURA_ADMIN_KEY": {
+            "opitem": "Vision Zero graphql-engine Endpoints",
+            "opfield": "production.Admin Key",
+            "opvault": vault_id,
+        },
+        "AWS_ACCESS_KEY_ID": {
+            "opitem": "CR3 Download IAM Access Key and Secret",
+            "opfield": "production.accessKeyId",
+            "opvault": vault_id,
+        },
+        "AWS_SECRET_ACCESS_KEY": {
+            "opitem": "CR3 Download IAM Access Key and Secret",
+            "opfield": "production.accessSecret",
+            "opvault": vault_id,
+        },
+        "AWS_DEFAULT_REGION": {
+            "opitem": "CR3 Download IAM Access Key and Secret",
+            "opfield": "production.awsDefaultRegion",
+            "opvault": vault_id,
+        },
+        "ATD_CRIS_CR3_URL": {
+            "opitem": "Vision Zero CRIS CR3 Download",
+            "opfield": "production.ATD_CRIS_CR3_URL",
+            "opvault": vault_id,
+        },
+        "AWS_CRIS_CR3_BUCKET_NAME": {
+            "opitem": "Vision Zero CRIS CR3 Download",
+            "opfield": "production.AWS_CRIS_CR3_BUCKET_NAME",
+            "opvault": vault_id,
+        },
+        "AWS_CRIS_CR3_BUCKET_PATH": {
+            "opitem": "Vision Zero CRIS CR3 Download",
+            "opfield": "production.AWS_CRIS_CR3_BUCKET_PATH",
+            "opvault": vault_id,
+        },
+    }
+
+    return onepasswordconnectsdk.load_dict(one_password_client, required_secrets)
+
+
+def process_crash_cr3_threaded(
+    crash_record, cris_browser_cookies, skipped_uploads_and_updates, verbose
+):
+    """Process a crash record in a separate thread."""
+    try:
+        process_crash_cr3(
+            crash_record, cris_browser_cookies, skipped_uploads_and_updates, verbose
+        )
+        if verbose:
+            print(f"Processed crash ID: {crash_record['crash_id']}")
+    except Exception as e:
+        print(f"Error processing crash ID {crash_record['crash_id']}: {str(e)}")
+        skipped_uploads_and_updates.append(str(crash_record["crash_id"]))
+
+
+def main():
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser(description="CRIS - CR3 Downloader")
+    parser.add_argument(
+        "-t",
+        "--threads",
+        type=int,
+        default=1,
+        help="Number of concurrent downloaders (default: 1)",
```

Comment:
so the default concurrent downloader is 1, but the docker-compose entrypoint overrides this to be 5

`entrypoint: /app/cr3_download.py -t 5`

wondering if the default should just be 5? or take the `-t 5` part off of the entrypoint?

Reply:
This is a great question Chia, thanks for asking about it and for catching that mismatch. I think there's no absolute right or wrong answer here, and fwiw, here's the reasoning that went into these choices. 

I think of the default in the program itself as being the default for when there is no contextual information about how the program is going to be used for what and by whom and that this default should represent the most conservative and safe choice, which is to just have one thread. 

Then, in the docker compose stack, I view that configuration as being in our own, specific context, and that it should reflect how we use the program. Under this thinking, that is where we can loosen up the configuration to a setting that is good for our particular use.

I realize that we don't really expect that there are other users of this software out there, and that it's open source for transparency and not for reuse, at least at this time, so I totally get that my reasoning is entirely conceptual and not practical, but at least that's what I was thinking when I picked those numbers.